docs/notebooks/active_learning.pct.py

```diff
@@ -0,0 +1,177 @@
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return {
+        "model": gpr,
+        "optimizer": gpflow.optimizers.Scipy(),
+        "optimizer_args": {
+            "minimize_args": {"options": dict(maxiter=100)},
+        },
+    }
+
+
+model = build_model(initial_data)
+
+# %% [markdown]
+# ## Active learning using predictive variance
+#
+# To do our first active learning example, we will use `PredictiveVariance`, one of the simplest active learning acquisition function. the `PredictiveVariance` sample a point on highest determinant of the predictive covariance of our model. Here, we will consider single query point using `PredictiveVariance`, the batch case will be shown later. We can utilize trieste's `BayesianOptimizer` to do the active learning steps.
+
+# %%
+from trieste.acquisition.optimizer import generate_continuous_optimizer
+from trieste.acquisition.rule import EfficientGlobalOptimization
+from trieste.acquisition.function import PredictiveVariance
+
+acq = PredictiveVariance()
+rule = EfficientGlobalOptimization(
+    builder=acq, optimizer=generate_continuous_optimizer(sigmoid=False)
+)
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+
+# %% [markdown]
+# To plot the contour of variance of our model at each step, we can set the `track_state` parameter to `True` in `bo.optimize()`, this will make trieste record our model at each iteration.
+
+# %%
+bo_iter = 5
+result = bo.optimize(bo_iter, initial_data, model, rule, track_state=True)
+
+# %% [markdown]
+# Then we can retrieve our final dataset from the active learning steps
+
+# %%
+dataset = result.try_get_final_dataset()
+query_points = dataset.query_points.numpy()
+observations = dataset.observations.numpy()
+
+# %% [markdown]
+# Finally, we can check the performance of our `PredictiveVariance` active learning acquisition function by plotting the predictive variance landscape of our model. We can see how it samples on regions with high variance value.
+
+# %%
+from util.plotting import plot_bo_points, plot_function_2d
+
+for i in range(bo_iter):
+
+    def pred_var(x):
+        _, var = result.history[i].models["OBJECTIVE"].model.predict_f(x)
+        return var
+
+    _, ax = plot_function_2d(
+        pred_var,
+        search_space.lower - 0.01,
+        search_space.upper + 0.01,
+        grid_density=20,
+        contour=True,
+        colorbar=True,
+        figsize=(10, 6),
+        title=["Variance contour with queried points at iter:" + str(i + 1)],
+        xlabel="$X_1$",
+        ylabel="$X_2$",
+    )
+    plot_bo_points(query_points[: num_initial_points + i], ax[0, 0], num_initial_points)
+
+
+# %% [markdown]
+# ## Batch active learning using predictive variance
+#
+# For batch active learning using `PredictiveVariance`, we can set the `num_query_points` with the size of batch we want on the `EfficientGLobalOptimization` rule. The rest step is the same with the single query point of `PredictiveVariance` acquisition function.
```

Comment:
For batch active learning, we must pass a `num_query_points` input to our `EfficientGLobalOptimization` rule.